apps/api/v2/src/types/index.ts

```diff
@@ -7,6 +7,9 @@ export interface PlatformApiLocals extends Record<string, unknown> {
     userId: number;
     teamId: number;
   };
+  user?: {
```

Comment:
I am thinkign that we need to split `PlatformApiLocals` into separate interfaces and then also have different `type Response` because some routes, like oAuth` will not have `apiKey` and vice versa and if we keep adding to `locals` then it will be hard to make sense of it.

What are your thoughts?

Reply:
yes, let's come up with a discriminative union